### PR TITLE
Small critical-section-related fixes.

### PR DESCRIPTION
**Pull request by Dirbaio**
See individual commit messages.

### CODE REVIEW

**Comment by adamgreig** [comment]
While we're here, could you add the feature to the "Optional features" section of the docs in `src/lib.rs` please?

**Comment by Dirbaio** [comment]
Done

**Review by thejpster — DISMISSED** [review]
LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -9,6 +9,14 @@
 //!
 //! # Optional features
 //!
+//! ## `critical-section-single-core`
+//!
+//! This feature enables a [`critical-section`](https://github.com/rust-embedded/critical-section)
+//! implementation suitable for single-core targets, based on disabling interrupt globally.
```
Comment (adamgreig) [inline src/lib.rs:15]:
```suggestion
//! implementation suitable for single-core targets, based on disabling interrupts globally.
```

File: src/lib.rs [hunk 2]
```diff
@@ -9,6 +9,14 @@
...
+//!
+//! It is **unsound** to enable it on multi-core targets. You should use a target-specific
+//! implementation that synchronizes all cores instead, typically provided by the HAL crate.
```
Comment (adamgreig) [inline src/lib.rs:18]:
```suggestion
//! It is **unsound** to enable it on multi-core targets or for code running in unprivileged mode,
//! and may cause functional problems in systems where some interrupts must be not be disabled
//! or critical sections are managed as part of an RTOS.
//! You should use a target-specific implementation instead, typically provided by a HAL crate.
```

File: src/macros.rs [hunk 3]
```diff
@@ -31,7 +31,10 @@ macro_rules! iprintln {
 /// at most once in the whole lifetime of the program.
 ///
 /// # Notes
-/// This macro is unsound on multi core systems.
+///
+/// This macro requires a `critical-section` implementation to be set. For single core systems, you can
+/// enable the `critical-section-single-core` feature for this crate. For multi core systems, you
+/// have to provide one from elsewhere, typically your chip's HAL crate.
```
Comment (adamgreig) [inline src/macros.rs:37]:
```suggestion
/// This macro requires a `critical-section` implementation to be set. For most single core systems,
/// you can enable the `critical-section-single-core` feature for this crate. For other systems, you
/// have to provide one from elsewhere, typically your chip's HAL crate.
```

**Review by adamgreig — CHANGES_REQUESTED** [review]
Few wording tweaks to highlight that even some single-core systems cannot use the feature, e.g. if unprivileged (see #233) or for things like softdevice or an RTOS.

**Review by adamgreig — APPROVED** [review]
thanks!

bors r+

**Comment by adamgreig** [comment]
bors retry

**Pull request merged**